docker-compose.yml

```
@@ -24,10 +24,15 @@ services:
       - ./dev/vault:/etc/vault
 
   db:
-    image: postgres:10.1
+    image: postgres:10.18
     ports:
       # 5432 may already in use by another PostgreSQL on host
       - "5433:5432"
+    environment:
+      POSTGRES_DB: warehouse
+      POSTGRES_PASSWORD: password
+    volumes:
+      - ./dev/example.sql.xz:/docker-entrypoint-initdb.d/example.sql.xz
```

```suggestion
      - ./dev/example.sql.xz:/docker-entrypoint-initdb.d/example.sql.xz:ro
```
While we're at it's, it's free.

---

Do we want to add a named volume with the DB data ? This should help have the DB ready more often.

---

> Do we also want to add a named volume with the DB data ? This should help have the DB ready more often.

I am not sure I understand why DB would be not ready in development process once initialized, and how named volume helps with that. For `make resetdb` it doesn't matter if it is a named volume or not.

---

Oh, I see. The named volume that will persist already imported data.

---

yes

---

I will address this in the thread below.

---

To elaborate: as of today, with an unnamed volume, if you delete (`docker-compose down`) and recreate your db container, the data is gone and you have to run `make initdb` (5 to 10 min) everytime. With a named volume, `docker-compose down` will leave the volume as-is, so the next time you run the stack, the DB is ready. If you need to reinit the db, you need to either delete the whole stack (`docker-compose down --volumes`) or delete the volume directly with `docker volume`.

---

Added the named volume.